### PR TITLE
[G7T5-70 & G7T5-71] Create LJ Deletion FE and BE

### PR DESCRIPTION
**Pull request by deadline2359**
# Description

Fixes [Jira ticket G7T5-70](https://is212g7t5.atlassian.net/browse/G7T5-70) and [Jira ticket G7T5-71](https://is212g7t5.atlassian.net/browse/G7T5-71)
- Created new endpoint where all courses under the learning journey ID would be deleted before deleting learning journey
- A popup for when staff deletes learning journey
- Link BE and FE

https://user-images.githubusercontent.com/89028059/194948116-a519c425-b066-4235-a15d-69d9061696b3.mov

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Migration

# How Has This Been Tested?
- [x] I have added the corresponding testcases into the [testcase document](https://docs.google.com/spreadsheets/d/1QOyUN_kFN0fddLkjAQz2DK3jou3cWdN9CJsNbl3v0Kg/edit#gid=0)

Please indicate the testcase ID you have added or are using

- [x] G7T5-61-1
- [x] G7T5-61-2

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/api/learningJourneyCourse.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    const res = await axiosLJCourseInstance.delete(`all/lj_id=${ljId}`);
+    if (res) {
+      return res.data;
+    }
+    throw new Error("No data returned from backend");
+  } catch (error) {
+    console.log(error);
+    return [];
+  }
+};
```
Comment (jonathantan1425) [inline app/src/api/learningJourneyCourse.js:22]: /nit would be good to add an EOF line here

File: app/src/components/learningJourney/details/LJDeletionPopUp.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+function DeleteLJButton({DeleteLJButtonClick}) {
+    return (
+        <button type="button" className="text-textColor bg-primaryColor hover:bg-secondaryColor font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2" onClick={DeleteLJButtonClick}>
+            Confirm
+        </button>        
+    )
+};
+
+export default LJDeletionPopUp;
```
Comment (jonathantan1425) [inline app/src/components/learningJourney/details/LJDeletionPopUp.jsx:46]: /nit same EOF line here as well

File: app/src/components/learningJourney/details/LJDeletionPopUpButton.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    return (
+        <div>
+            <button type="button" className="text-callToActionColor1" onClick={onClick}>delete?</button>
+            {isOpen && <LJDeletionPopUp ljId={ljId} />}
+        </div>
+    );
+};
+
+
+export default LJDeletionPopUpButton;
```
Comment (jonathantan1425) [inline app/src/components/learningJourney/details/LJDeletionPopUpButton.jsx:17]: /nit same EOF line here as well

File: app/src/api/learningJourneyCourse.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const axiosLJCourseInstance = axios.create({
+  baseURL: LEARNING_JOURNEY_COURSE_ENDPOINT,
+  timeout: 5000,
+  headers: { "X-Custom-Header": "foobar" },
+});
+
+
+export const deleteLearningJourneyCourseWithLJId = async (ljId) => {
+  try {
+    const res = await axiosLJCourseInstance.delete(`all/lj_id=${ljId}`);
```
Comment (jonathantan1425) [inline app/src/api/learningJourneyCourse.js:13]: Is there an endpoint that takes in the path `all/lj_ud=${ljId}` here? I don't remember the fastapi endpoints to be formatted with a query param looking path variable but without the `?` notation

File: app/src/components/learningJourney/details/LJDeletionPopUp.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    const onClick = () => {setIsOpen(!isOpen)}
+
+    if(!isOpen){
+        return null
+    }
+
+    const DeleteLJButtonClick = (e) => {
+        e.stopPropagation();
+        const res = deleteLearningJourneyWithLJId(ljId);
+        console.log(res);
```
Comment (jonathantan1425) [inline app/src/components/learningJourney/details/LJDeletionPopUp.jsx:16]: /nit remove if no longer necessary

File: app/src/api/learningJourney.js [hunk 6]
```diff
@@ -47,6 +48,28 @@ export const createLearningJourneyWithJobId = async (jobId) => {
   }
 };
 
+export const deleteLearningJourneyWithLJId = async (ljId) => {
+  try {
+    const res1 = deleteLearningJourneyCourseWithLJId(ljId);
```
Comment (quinncheong) [inline app/src/api/learningJourney.js:53]: Best to rename res1 and res2 better

File: app/src/components/learningJourney/DropdownButton.jsx [hunk 7]
```diff
@@ -1,7 +1,8 @@
 import React from "react";
 import "flowbite";
+import LJDeletionPopUpButton from "./details/LJDeletionPopUpButton";
 
-function DropdownButton() {
+function DropdownButton(ljId) {
```
Comment (quinncheong) [inline app/src/components/learningJourney/DropdownButton.jsx:5]: LJ in caps

File: app/src/components/learningJourney/LearningJourneyTile.jsx [hunk 8]
```diff
@@ -8,7 +8,7 @@ function LearningJourneyTile({ LJId, jobName, jobDesc, isJobActive }) {
 
   return (
     <div className='max-w-sm bg-white rounded-lg border border-gray-100 hover:shadow-lg hover:shadow-orange-200 mt-10 mx-auto'>
-      <DropdownButton />
+      <DropdownButton ljId={LJId} />
```
Comment (quinncheong) [inline app/src/components/learningJourney/LearningJourneyTile.jsx:11]: LJ in caps

File: app/src/components/learningJourney/details/LJDeletionPopUp.jsx (hunk 5 above)
Comment (quinncheong) [reply app/src/components/learningJourney/details/LJDeletionPopUp.jsx:16]: Seems necessary. Remember that functions should be camel case so deleteLJ...

**Review by quinncheong — CHANGES_REQUESTED** [review]

**Comment by quinncheong** [comment]
Remember to name your PR with the ticket, incl image if possible, and resolve comments once fixed.

**Review by jonathantan1425 — APPROVED** [review]

**Review by quinncheong — APPROVED** [review]

**Pull request merged**